src/rgw/rgw_rest.cc

```diff
@@ -1486,70 +1486,59 @@ static int read_all_chunked_input(req_state *s, char **pdata, int *plen, const u
 	need_to_read *= 2;
 
       if ((unsigned)total > max_read) {
-	free(data);
-	return -ERANGE;
+	return std::make_tuple(-ERANGE, std::move(pdata), 0);
       }
       total += need_to_read;
-
-      void *p = realloc(data, total + 1);
-      if (!p) {
-	free(data);
-	return -ENOMEM;
-      }
-      data = (char *)p;
     } else {
       break;
     }
-
   } while (true);
-  data[len] = '\0';
+  data.push_back('\0');
 
-  *pdata = data;
-  *plen = len;
+  pdata = std::make_unique<char[]>(len + 1);
+  strcpy(pdata.get(), data.data());
```

Comment:
if we have to allocate again and copy to return a unique_ptr, i'm not sure that it's worth using vector internally. what do you think about having these functions return the vector\<char\> instead of unique_ptr?

vector knows its own size, so you wouldn't have to return it separately in the tuple

Reply:
is the problem unique_ptr or make_unique?

Reply:
the problem is that 'data' here is a vector, with its own memory management, and we're returning it as a unique_ptr; doing that requires a separate allocation

Reply:
I didn't expect we'd cease to deal with this as a sequence of bytes, and in that case, unique_ptr would be fine